docs/API_KEY.md

```diff
@@ -0,0 +1,151 @@
+# How to add new/own API key to tinode?
+
+## What is API key?
+
+Trinode encrypts data for security. For this purpose, two keys are used:
+
+* `HMAC salt` in server side. Hereafter referred to as _Server Key_.
+* `API key` in client side. Hereafter referred to as _Client Key_.
+
+## Generate Server/Client Keys
+
+### 1) Generate Automatically
+
+Use the [keygen](https://github.com/tinode/chat/blob/master/keygen/README.md) which is available in the tinode folder. Run this program with the `-salt auto` option. As below:
+
+```bash
+./keygen -salt auto
+```
+
+result like:
+
+```text
+API key v1 seq1 [ordinary]: AQAAAAABAACGOIyP2vh5avSff5oVvMpk
+Used HMAC salt: TC0Jzr8f28kAspXrb4UYccJUJ63b7CSA16n1qMxxGpw=
+```
+
+There are two lines:
+
+* In first line detect _Client Key_. In this example is `AQAAAAABAACGOIyP2vh5avSff5oVvMpk`.
+* In second line detect _Server Key_. In this example is `TC0Jzr8f28kAspXrb4UYccJUJ63b7CSA16n1qMxxGpw=`.
+
+*Note*: This value is random and will be different for you.
+*Note*: You can use the `-sequence` option to make the key more complex. like:
+
+```bash
+./keygen -salt TC0Jzr8f28kAspXrb4UYccJUJ63b7CSA16n1qMxxGpw= -sequence 5
+```
+
+result is:
+
+```text
+API key v1 seq5 [ordinary]: AQAAAAAFAAAqXLPhti0JwhyVXBWKXTPW
+Used HMAC salt: TC0Jzr8f28kAspXrb4UYccJUJ63b7CSA16n1qMxxGpw=
+```
+
+
+### 2) Generate Manually

```

Comment:
I'm not sure this section is needed at all. I will probably remove the ability to use external salt as it brings no value. It's there for historical reasons only.